include/mxnet/ndarray.h

```diff
@@ -35,12 +35,12 @@
 #include <map>
 #include <string>
 #include <memory>
+#if MXNET_USE_MKLDNN == 1
+#include <mkldnn.hpp>
```

Comment:
Does this need to be installed independently or is it a submodule in '3rdparty', for instance?

Reply:
I'm not sure. Both seem to work fine. Intel provides a python script to install MKLDNN in the system.

Reply:
it can be compiled from source right? we should make it a submodule

Reply:
I did a PR on mkl-dnn a few months back to allow it to be included as a submodule, so it shoulkd be able to work ok from cmake: https://github.com/01org/mkl-dnn/pull/44